src/Application.php

```diff
@@ -1,29 +1,31 @@
 <?php
 namespace Jalno\Lumen;
 
-use Jalno\Storage\Providers\StorageProvider;
+use Jalno\Lumen\Contracts\IPackages;
 use Laravel\Lumen\Application as ParentApplication;
 use Jalno\AutoDiscovery\Providers\AutoDiscoverProvider;
 use Illuminate\Contracts\Console\Kernel as ConsoleKernelContract;
 
 class Application extends ParentApplication
 {
-    protected $packagePath;
+    public IPackages $packages;
+
     /**
      * Create a new Lumen application instance.
      *
      * @param  string|null  $basePath
-     * @param class-string<Contracts\IPackage>
+     * @param class-string<Contracts\IPackage> $primaryPackage
+     * @param IPackages $packages
      * @return void
      */
-    public function __construct($basePath = null, string $primaryPackage)
+    public function __construct($basePath, string $primaryPackage, IPackages $packages)
     {
+        $this->packages = $packages;
```

Comment:
It's a service in service provider.
We can't construct it, it's ServiceContainer job.